wicket-core/src/main/java/org/apache/wicket/ajax/AjaxOnDomReadyClientInfoBehavior.java

```diff
@@ -0,0 +1,124 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.wicket.ajax;
+
+import org.apache.wicket.Component;
+import org.apache.wicket.Session;
+import org.apache.wicket.ajax.attributes.AjaxRequestAttributes;
+import org.apache.wicket.markup.head.IHeaderResponse;
+import org.apache.wicket.markup.head.JavaScriptHeaderItem;
+import org.apache.wicket.markup.head.OnDomReadyHeaderItem;
+import org.apache.wicket.markup.html.pages.BrowserInfoForm;
+import org.apache.wicket.protocol.http.request.WebClientInfo;
+import org.apache.wicket.request.IRequestParameters;
+import org.apache.wicket.request.cycle.RequestCycle;
+import org.apache.wicket.util.lang.Args;
+import org.danekja.java.util.function.serializable.SerializableBiConsumer;
+
+/**
+ * A behavior that collects the information to populate
+ * WebClientInfo's ClientProperties by using Ajax. Compared to
+ * {@link AjaxClientInfoBehavior} this class does not use a timer
+ * but the DOM ready "event" to collect browser info.
+ *
+ * @see #onClientInfo(AjaxRequestTarget, WebClientInfo)
+ */
+public class AjaxOnDomReadyClientInfoBehavior extends AbstractDefaultAjaxBehavior
+{
+	private static final long serialVersionUID = 1L;
+
+	/**
+	 * Constructor.
+	 *
+	 */
+	public AjaxOnDomReadyClientInfoBehavior()
+	{
+		super();
+	}
+
+	@Override
+	protected void respond(AjaxRequestTarget target)
+	{
+		RequestCycle requestCycle = RequestCycle.get();
+
+		IRequestParameters requestParameters = requestCycle.getRequest().getRequestParameters();
+		WebClientInfo clientInfo = newWebClientInfo(requestCycle);
+		clientInfo.getProperties().read(requestParameters);
+		Session.get().setClientInfo(clientInfo);
+
+		onClientInfo(target, clientInfo);
+	}
+
+	protected WebClientInfo newWebClientInfo(RequestCycle requestCycle)
+	{
+		return new WebClientInfo(requestCycle);
+	}
+
+	/**
+	 * A callback method invoked when the client info is collected.
+	 * 
+	 * @param target
+	 *          The Ajax request handler
+	 * @param clientInfo
+	 *          The collected info for the client 
+	 */
+	protected void onClientInfo(AjaxRequestTarget target, WebClientInfo clientInfo)
+	{
+	}
+
+	@Override
+	protected void updateAjaxAttributes(AjaxRequestAttributes attributes)
+	{
+		super.updateAjaxAttributes(attributes);
+		attributes.setMethod(AjaxRequestAttributes.Method.POST);
```

Comment:
Let's be explicit here and call `attributes.setEventNames("domready")`.
Currently it depends on an implementation [detail](https://github.com/apache/wicket/blob/8e33df5dae79d3ba7dbf4ffa2115ddf6b51518eb/wicket-core/src/main/java/org/apache/wicket/ajax/res/js/wicket-ajax-jquery.js#L1666).